### PR TITLE
add RethinkDB as a plugin

### PR DESCRIPTION
**Pull request by jipperinbham**
I don't have a tests yet so feedback there would be great as well as any overall feedback.

### CODE REVIEW

File: plugins/rethinkdb/rethinkdb.go [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+		Address:       serv.Host,
+		DiscoverHosts: false,
+	}
+	if serv.User != nil {
+		pwd, set := serv.User.Password()
+		if set && pwd != "" {
+			connectOpts.AuthKey = pwd
+		}
+	}
+	r.session, err = gorethink.Connect(connectOpts)
```
Comment (evanphx) [inline plugins/rethinkdb/rethinkdb.go]: You don't close `r.session` here (or close a previous one) so you'll open infinite new connections.

**Comment by evanphx** [comment]
I haven't used RethinkDB so I can't comment on if this is the right API or info to gather, I'll have to trust you there.

You'll need to manage that `r.session` properly though. If the session is expensive to setup, you can hold it across `Gather` calls, but you'll need to make sure it's still alive before using it.

**Comment by jipperinbham** [comment]
@evanphx I added a `defer` to close the session after each Gather for now but if it looks to be an issue constantly opening/closing the session with each invocation, I'll adjust.

**Comment by jipperinbham** [comment]
@evanphx I've updated this PR with some reorg of the plugin, added tests, and squashed + rebased from master. 

Any additional feedback?

**Pull request closed without merging**